test/functional/session_leak_test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const core = require('mongodb-core');
+const MongoClient = require('../../lib/mongo_client');
+const ServerSessionPool = core.Sessions.ServerSessionPool;
+
+(() => {
```

Comment:
why is this wrapped in an IIFE?

Reply:
I don't think it actually have to be. I'll remove it.

Reply:
I'm not sure A+ matters anymore now that this is part of the spec. Take a look here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Specifications, everything that supports Promises supports `catch`

Reply:
Ok, I'll change that :)